Lib/test/test_str.py

```diff
@@ -2589,7 +2589,8 @@ def test_compare(self):
 
     def test_free_after_iterating(self):
         support.check_free_after_iterating(self, iter, str)
-        support.check_free_after_iterating(self, reversed, str)
+        if not sys._is_gil_enabled():
```

Comment:
In the free-threading build we do not free the sequence directly after iterating to avoid race conditions. We therefore skip this test.